src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/Helpers/TeddyHelper.cs

```diff
@@ -221,9 +221,9 @@ public static Vector128<byte> LoadAndPack16AsciiChars(ref char source)
             {
                 return Sse2.PackUnsignedSaturate(source0.AsInt16(), source1.AsInt16());
             }
-            else if (AdvSimd.IsSupported)
```

Comment:
Do we want to keep this case for when AdvSimd.IsSupported but !AdvSimd.Arm64.IsSupported?

Reply:
In this case they both represent the "is ARM with SIMD" case and are effectively interchangeable.
We should never reach this path if `AdvSimd.Arm64` isn't supported, so the fallback to `ThrowUnreachableException` is correct.
https://github.com/dotnet/runtime/blob/a1667657373d846df5c8df8fc613a6d709f89957/src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/StringSearchValues.cs#L130-L131